python/tests/test_phylo_formats.py

```diff
@@ -146,3 +147,83 @@ def test_single_node_label(self):
             [labels[tree.root]] + [None for _ in range(len(list(tree.nodes())) - 1)],
         )
+
+
+class TestNexus(unittest.TestCase):
+    """
+    Tests that the nexus output has the properties that we need using
+    external Nexus parser.
+    """
+
+    random_seed = 155
+
+    # TODO: remove manual curation of leaf_labels when creating nexus output
+    def verify_tree(self, nexus_tree, tree, inv_labels):
+        root = tree.root
+        self.assertEqual(len(nexus_tree.get_terminals()), len(inv_labels))
+        for u in nexus_tree.get_terminals():
+            t_id = inv_labels[nexus_tree.node(u).data.taxon]
+            self.assertTrue(tree.is_leaf(t_id))
+            node = nexus_tree.node(u)
+            while t_id != root:
+                self.assertAlmostEqual(node.data.branchlength, tree.branch_length(t_id))
+                ancestor_id = node.get_prev()
+                node = nexus_tree.node(ancestor_id)
+                t_id = tree.parent(t_id)
+            self.assertIsNone(node.get_prev())
+
+    def verify_nexus_topology(self, treeseq, node_labels=None):
+        nexus = treeseq.nexus(precision=16)
+        if node_labels is None:
+            leaf_labels = {u: str(u + 1) for u in treeseq.samples()}
+        else:
+            leaf_labels = {u: node_labels[u] for u in treeseq.samples()}
+        inv_labels = {v: k for k, v in leaf_labels.items()}
+
+        nexus_treeseq = Nexus.Nexus(nexus)
+
+        interval = [0.0, 0.0]
+        for tree, nexus_tree in zip(treeseq.trees(), nexus_treeseq.trees):
+            name = nexus_tree.name
+            split_name = name.split("_")
+            self.assertEqual(len(split_name), 2)
+            start = float(split_name[0][4:])
+            end = float(split_name[1])
+            self.assertAlmostEqual(interval[1], start)
+            interval[0] = start
+            interval[1] = end
+
+            self.verify_tree(nexus_tree, tree, inv_labels)
+        self.assertAlmostEqual(interval[1], 1)
+
+    def get_binary_example(self):
+        ts = msprime.simulate(
+            sample_size=4, recombination_rate=2, random_seed=self.random_seed
+        )
+        return ts
+
+    def test_binary_tree(self):
+        ts = self.get_binary_example()
+        self.verify_nexus_topology(ts)
+
+    def get_nonbinary_example(self):
+        ts = msprime.simulate(
+            sample_size=5,
+            recombination_rate=10,
+            random_seed=self.random_seed,
+            demographic_events=[
+                msprime.SimpleBottleneck(time=0.5, population=0, proportion=1)
+            ],
+        )
+        # Make sure this really has some non-binary nodes
+        found = False
+        for e in ts.edgesets():
+            if len(e.children) > 2:
+                found = True
+                break
+        self.assertTrue(found)
+        return ts
+
+    def test_nonbinary_example(self):
+        ts = self.get_nonbinary_example()
+        self.verify_nexus_topology(ts)
```

Comment:
We also want to test the single tree example:
```
def test_single_tree(self):
     ts = msprime.simulate(10, random_seed=2)
     self.verify_nexus_topology(ts)
```

It might be nice to think about how we can avoid duplicating code within this test module. Perhaps we could do a test mixin, or even just factor out the ``get_X_example`` functions. We do want a few more test examples in any case.